src/commands/dev/config_secrets_edit.rs

```diff
@@ -0,0 +1,213 @@
+    edited_secrets: &HashMap<String, String>,
+) -> Vec<(String, String)> {
+    let max_key_len = edited_secrets
+        .values()
+        .map(|key| key.len())
+        .max()
+        .unwrap_or(0);
+    let max_old_value_len = secrets.values().map(|key| key.len()).max().unwrap_or(0);
+
+    let green = Style::new().for_stderr().green();
+    let red = Style::new().for_stderr().red();
+
+    let mut items: Vec<(String, String)> = edited_secrets
+        .iter()
+        .map(|(key, new_value)| {
+            let mut key_with_style: String = key.clone();
+            let mut old_value_with_style: String = new_value.clone();
+            let mut new_value_with_style: String = new_value.clone();
+
+            // Insert Key:
+            if !secrets.contains_key(key) {
+                key_with_style = format!("{}{}", green.apply_to("+"), green.apply_to(key));
+            } else {
+                // Updated values:
+                let old_value = secrets.get(key).unwrap_or(&String::from("")).to_string();
+                let changeset = Changeset::new(&old_value, new_value, "");
+
+                let changes: Vec<String> = changeset
+                    .diffs
+                    .iter()
+                    .map(|diff| match diff {
+                        Difference::Same(s) => s.to_string(),
+                        Difference::Add(s) => format!("{}", green.apply_to(s)),
+                        Difference::Rem(s) => format!("{}", red.apply_to(s)),
+                    })
+                    .collect();
+
+                new_value_with_style = changes.join("");
+                old_value_with_style = old_value;
+            }
+
+            // Return the item:
+            (
+                format!(
+                    "{:<width$} \t {} → {}",
+                    key_with_style,
+                    format_args!(
+                        "{:<width$}",
+                        old_value_with_style,
+                        width = max_old_value_len
+                    ),
+                    new_value_with_style,
+                    width = max_key_len
+                ),
+                key.to_owned(),
+            )
+        })
+        .collect();
+
+    // Handle deleted keys:
+    let deleted_keys = secrets
+        .keys()
+        .filter(|key| !edited_secrets.contains_key(*key))
+        .collect::<Vec<&String>>();
+
+    for key in deleted_keys {
+        let item = (
+            format!(
+                "{:<width$} \t {} → {}",
+                format!("{}{}", style("-").red(), style(key).red()),
+                format_args!(
+                    "{:<width$}",
+                    secrets.get(key).unwrap_or(&String::from("")),
+                    width = max_old_value_len
+                ),
+                secrets.get(key).unwrap_or(&String::from("")),
+                width = max_key_len
+            ),
+            key.to_owned(),
+        );
+
+        items.push(item);
+    }
+
+    items
+}
+
+#[cfg(test)]
```

Comment:
We should have a test to check the case of not able to find any annotation(s) at all. In that case we should display a message to say that `Not able to detecting any annotations, please refer to this guideline [url] to configure your config before using this command`.